xdsl/xdsl_opt_main.py

```diff
@@ -212,9 +212,21 @@ def setup_pipeline(self):
             if p.name not in self.available_passes:
                 raise Exception(f"Unrecognized pass: {p.name}")
 
-        self.pipeline = [
-            self.available_passes[p.name].from_pass_spec(p) for p in pipeline
-        ]
+        def callback(
+            previous_pass: ModulePass, module: ModuleOp, next_pass: ModulePass
+        ) -> None:
+            if not self.args.disable_verify:
+                module.verify()
```

Comment:
Here, we don't verify after the last pass anymore?

Reply:
Ah yes that's true, I'll add the verification back

Reply:
Done